### PR TITLE
SignInWithGithub

### PR DESCRIPTION
**Pull request by shreya-mishra**
**Ticket**
https://github.com/Real-Dev-Squad/mobile-app/issues/199

https://github.com/Real-Dev-Squad/mobile-app/assets/61110378/c875dc7e-462a-4e31-b76a-9be7e9aa8777

### CODE REVIEW

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 1]
```diff
@@ -24,7 +25,10 @@ import { urls } from '../../constants/appConstant/url';
 import AuthApis from '../../constants/apiConstant/AuthApi';
 import { CameraScreen } from 'react-native-camera-kit';
 import CustomModal from '../../components/Modal/CustomModal';
+import { githubConfig } from '../../../config/config';
 
+const githubAuthUrl =
+  'https://github.com/login/oauth/authorize?client_id=YOUR_CLIENT_ID&redirect_uri=yourappname://oauth&scope=user';
```
Comment (biswanath-tewari) [inline src/screens/AuthScreen/AuthScreen.tsx]: placeholders needs to be replaced with actual values

**Review by biswanath-tewari — DISMISSED** [review]

File: android/app/src/main/java/com/rdsapp/MainActivity.java [hunk 2]
```diff
@@ -12,4 +16,13 @@ public class MainActivity extends ReactActivity {
...
     return "RDSApp";
   }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    // ATTENTION: This was auto-generated to handle app links.
+    Intent appLinkIntent = getIntent();
+    String appLinkAction = appLinkIntent.getAction();
+    Uri appLinkData = appLinkIntent.getData();
```
Comment (prudhvir3ddy) [inline android/app/src/main/java/com/rdsapp/MainActivity.java:26]: Not seeing any usage of this. 

I hope react native linking APIs provides this data in typescript files itself 

if it is then we can remove this code

Comment (shreya-mishra) [reply android/app/src/main/java/com/rdsapp/MainActivity.java:26]: This is automatically linked while implementing app links deeplinking process.


Doesnt matter for now I guess nothing is working as expected

File: config/config.js [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+//     authorizationEndpoint: 'https://github.com/login/oauth/authorize',
+//     tokenEndpoint: 'https://github.com/login/oauth/access_token',
+//     revocationEndpoint:
+//       'https://github.com/settings/connections/applications/969f022b19e148218c41'
+//   },
+// };
+
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+export const clientId = '23c78f66ab7964e5ef97';
```
Comment (prudhvir3ddy) [inline config/config.js:3]: with the given client ID, i am seeing the callback URL is 
`https://api.realdevsquad.com/auth/github/callback`

check same in developer settings of the GitHub 

we are not handling this URL in the app and path in AndroidManifest.xml 

we need to handle this path

**Comment by ashifkhn** [comment]
I have installed the app locally from the branch it keeps loading after signing in. It redirects me to the app. So the redirection is working fine.

**Comment by shreya-mishra** [comment]
> I have installed the app locally from the branch it keeps loading after signing in. It redirects me to the app. So the redirection is working fine.

check the recent code ! it worked fine now.

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**